File: src/backend/src/Modules/Cart/Ecommerce.Modules.Cart.Application/Models/CartModels.cs
using Ecommerce.Modules.Cart.Domain.Enums;

namespace Ecommerce.Modules.Cart.Application.Models;

public sealed record CartItemResponse(
    Guid Id,
    Guid ProductId,
    string ProductName,
    string ProductSlug,
    string Sku,
    int Quantity,
    decimal UnitPrice,
    decimal LineTotal,
    bool IsAvailable,
    int AvailableStock);

public sealed record CartResponse(
    Guid Id,
    CartStatus Status,
    IReadOnlyList<CartItemResponse> Items,
    decimal Subtotal,
    int TotalItems,
    DateTime CreatedAt,
    DateTime? UpdatedAt);

File: src/frontend/src/app/core/models/cart.models.ts
export interface CartItem {
  id: string;
  productId: string;
  productName: string;
  productSlug: string;
  sku: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
  isAvailable: boolean;
  availableStock: number;
}

export interface Cart {
  id: string;
  status: string;
  items: CartItem[];
  subtotal: number;
  totalItems: number;
  createdAt: string;
  updatedAt?: string | null;
}
